File: Backend/RealManager.Services/TeamService.cs
using RealManager.Domain;
using RealManager.Domain.Enums;
using RealManager.Repositories.Interfaces;
using RealManager.Services.Interfaces;
using Services.Interfaces;

namespace RealManager.Services
{
    public class TeamService : ITeamService
    {
        private readonly ITeamRepository _teamRepository;
        private readonly IPlayerService _plyerService;
        public TeamService(ITeamRepository teamRepository, IPlayerService plyerService)
        {
            _teamRepository = teamRepository;
            _plyerService = plyerService;
        }

        public Team CreateTeam(Team team){
            return _teamRepository.Create(team);
        }

        public Team CreateRandomTeam(string teamName)
        {
            Team newTeam = new Team
            {
                Name = teamName
            };

            for (int i=0; i < 23; i++)
            {
                Player newPlayer;
                if(i == 0)
                {
                    newPlayer = _plyerService.CreateRandomPlayer(Position.GK, newTeam.Id);
                    newTeam.Players.Add(newPlayer);
                    newTeam.Starters.Add(newPlayer);
                }
                else if (i >= 1 && i <= 4)
                {
                    newPlayer = _plyerService.CreateRandomPlayer(Position.DF, newTeam.Id);
                    newTeam.Players.Add(newPlayer);
                    newTeam.Starters.Add(newPlayer);
                }
                else if (i >= 5 && i <= 7)
                {
                    newPlayer = _plyerService.CreateRandomPlayer(Position.MF, newTeam.Id);
                    newTeam.Players.Add(newPlayer);
                    newTeam.Starters.Add(newPlayer);
                }
                else if (i >= 8 && i <= 10)
                {
                    newPlayer = _plyerService.CreateRandomPlayer(Position.ATA, newTeam.Id);
                    newTeam.Players.Add(newPlayer);
                    newTeam.Starters.Add(newPlayer);
                }
                else
                {
                    newPlayer = _plyerService.CreateRandomPlayer(teamId: newTeam.Id);
                    newTeam.Players.Add(newPlayer);
                }
            }

            _teamRepository.Create(newTeam);

            return newTeam;
        }
    }
}

File: BackendNode/services/teamService.js
const Team = require('../domain/team');
const Position = require('../domain/position');
const Player = require('../domain/player');

const TeamRepository = require('../data/teamRepository');
const teamRepository = new TeamRepository();

const {v4: uuidv4} = require('uuid');
const constants = require('../util/constantes');


class TeamService {
    async createRandomTeam (teamName) {
        const newTeam = new Team();
        newTeam.name = teamName;
        newTeam._id = uuidv4();

        for(let i = 0; i < 23; i++) {
            let newPlayer;
            if (i === 0) {
                newPlayer = this.createRandomPlayer(Position.GK, newTeam._id);
                newTeam.players.push(newPlayer);
                newTeam.starters.push(newPlayer);
            }
            else if (i >= 1 && i <= 4) {
                newPlayer = this.createRandomPlayer(Position.DF, newTeam._id);
                newTeam.players.push(newPlayer);
                newTeam.starters.push(newPlayer);
            }
            else if (i >= 5 && i <= 7) {
                newPlayer = this.createRandomPlayer(Position.DF, newTeam._id);
                newTeam.players.push(newPlayer);
                newTeam.starters.push(newPlayer);
            }
            else if (i >= 8 && i <= 10) {
                newPlayer = this.createRandomPlayer(Position.DF, newTeam._id);
                newTeam.players.push(newPlayer);
                newTeam.starters.push(newPlayer);
            }
            else {
                newPlayer = this.createRandomPlayer(undefined, newTeam._id);
                newTeam.players.push(newPlayer);
            }
        }

        const createdTeam = await teamRepository.createTeam(newTeam);

        return createdTeam;
    }

    async getTeam(teamId) {
        var team = await teamRepository.getTeamById(teamId);
        return team;
    }

    createRandomPlayer(position, newTeamId) {
        const newPlayer = new Player();
        newPlayer._id = uuidv4();

        const indexFirstName = Math.floor((Math.random() * constants.possibleNames.length));
        const indexSurname = Math.floor((Math.random() * constants.possibleSurnames.length));

        newPlayer.name = constants.possibleNames[indexFirstName] + ' ' + constants.possibleSurnames[indexSurname];

        newPlayer.pace = Math.floor((Math.random() * 100) + 1);
        newPlayer.pass = Math.floor((Math.random() * 100) + 1);
        newPlayer.physical = Math.floor((Math.random() * 100) + 1);
        newPlayer.defence = Math.floor((Math.random() * 100) + 1);
        newPlayer.shoot = Math.floor((Math.random() * 100) + 1);
        newPlayer.drible = Math.floor((Math.random() * 100) + 1);
        newPlayer.overall = [newPlayer.pace, newPlayer.pass, newPlayer.physical, newPlayer.defence, newPlayer.shoot, newPlayer.drible].reduce((a, b) => a + b) / 6;

        newPlayer.position = position ? position : this.selectRandomPosition();

        newPlayer.teamId = newTeamId;

        return newPlayer
    }

    selectRandomPosition() {
        var keys = Object.keys(Position);
        return Position[keys[ keys.length * Math.random() << 0]];
    }
}

module.exports = TeamService;